mlir/lib/IR/OperationSupport.cpp

```diff
@@ -798,15 +844,24 @@ OperationEquivalence::isRegionEquivalentTo(Region *lhs, Region *rhs,
     return false;
 
   // 2. Compare operands.
-  for (auto operandPair : llvm::zip(lhs->getOperands(), rhs->getOperands())) {
-    Value curArg = std::get<0>(operandPair);
-    Value otherArg = std::get<1>(operandPair);
-    if (curArg == otherArg)
-      continue;
-    if (curArg.getType() != otherArg.getType())
-      return false;
-    if (failed(checkEquivalent(curArg, otherArg)))
+  if (checkCommutativeEquivalent &&
+      lhs->hasTrait<mlir::OpTrait::IsCommutative>()) {
+    auto lhsRange = lhs->getOperands();
+    auto rhsRange = rhs->getOperands();
+    if (failed(checkCommutativeEquivalent(lhsRange, rhsRange)))
```

Comment:
Do we actually need `checkCommutativeEquivalent`? Can we do something like this if the op has the `IsCommutative` trait?

```c++
SmallVector<Value> rhsOperands(rhs->getOperands().begin(), rhs->getOperands().end());
for (Value lhsOperand : lhs->getOperands()) {
  auto it = llvm::find_if(rhsOperands, [&](Value rhsOperand) {
      return succeeded(checkEquivalent(lhsOperand, rhsOperand));
  });
  if (it == rhsOperands.end()) {
    // Could not find equivalent operand.
    return false;
  }
  rhsOperands.erase(it);
}
assert(rhsOperands.empty());
```

Reply:
We could, but this is O(n^2) vs O(n log n). Now, this should be replaced by the explicit commutative sort pass in pipeline so that here it should suffice to just do the linear scan. The default param means its easy to drop again at that point without needing to update.